public/app/plugins/panel/barchart/BarChart.tsx

```diff
@@ -27,6 +27,9 @@ export const BarChart: React.FC<BarChartProps> = (props) => {
   const theme = useTheme2();
   const { eventBus } = usePanelContext();
 
+  const frame0Ref = useRef<DataFrame>();
+  frame0Ref.current = props.frames[0];
```

Comment:
surely, this is a hack?. :shrug: 

we need the `rawValue` callback below to always have a fresh `frames[0]` reference.